nncf/torch/pruning/filter_pruning/algo.py

```diff
@@ -253,12 +256,36 @@ def _init_pruned_modules_params(self):
             self.pruning_quotas[cluster.id] = np.floor(self._modules_out_channels[cluster.elements[0].node_name] \
                                                        * self.pruning_quota)
 
+    def _calculate_output_shape(self, graph: NNCFGraph, node: NNCFNode) -> Tuple[int, ...]:
```

Comment:
Can we have a test for this case?

I think we will have more such cases in the future. Is it possible to introduce a logic that we check the connectivity of the NNCF graph and raise an exception (or maybe a warning) if there are problems?

Reply:
Yes, will add tests.

I think add warning before use _calculate_output_shape, with message that node have output edge. ​
Exception was before on line `out_edge = graph.get_output_edges(node)[0]` with out of index.

> Is it possible to introduce a logic that we check the connectivity

It's possible, but we need discussion (what logic in nncf it can affect? what we can do with disconnected graph? list of test cases). I will create issue about it.

Reply:
Besides creating an issue, can you also think about the proposal?